### ISSUE TITLE
incompatible with rails 7.2

### ISSUE THREAD

**Issue by ssugiyama** [carrierwaveuploader/carrierwave#2751]
carrierwave version: 3.0.7

In production environment the following error occurs.

```ruby
% RAILS_ENV=production rails s
=> Booting Puma
=> Rails 7.2.0 application starting in production 
=> Run `bin/rails server --help` for more startup options
Exiting
/usr/local/rvm/gems/default/gems/carrierwave-3.0.7/lib/carrierwave/uploader/configuration.rb:163:in `fog_provider=': private method `warn' called for class ActiveSupport::Deprecation (NoMethodError)
        from /workspaces/yoka-server/api_root/config/initializers/carrierwave.rb:5:in `block in <main>'
        from /usr/local/rvm/gems/default/gems/carrierwave-3.0.7/lib/carrierwave/uploader/configuration.rb:177:in `configure'
        from /usr/local/rvm/gems/default/gems/carrierwave-3.0.7/lib/carrierwave.rb:14:in `configure'
        from /workspaces/yoka-server/api_root/config/initializers/carrierwave.rb:3:in `<main>'
(snip)
```

**Comment by nilshaebel** [comment]
In my case _CarrierWave.deprecator.warn_ has been called due to having a (deprecated) _extension_whitelist_ method in my Uploader. Renaming the method to _extension_allowlist_ fixed the issue for me.

**Example:**

In _app/uploaders/foo_uploader.rb_, replace:
```
def extension_whitelist
  %w(...)
end
```

with:
```
def extension_allowlist
  %w(...)
end```

**Comment by ssugiyama** [comment]
I have no methods named xxx_whitelist nor xxx_blacklist, but the error occurs.

**Comment by mshibuya** [comment]
Fixed by https://github.com/carrierwaveuploader/carrierwave/commit/b25477d4dc2b32ee3da896714b081e6123737fa3, which is included in the latest release 3.1.0.beta.
